azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -126,7 +126,7 @@ def on_hold(self):
         raise NotImplementedError()
 
     @property
-    def agent_manifests(self):
+    def agent_families(self):
```

Comment:
There was a class in restapi.py named VMAgentManifest which did not represent a manifest, but rather the GAFamilies element in ExtensionConfigs (which includes the name of the family (Prod vs Test), the required agent version, and the URIs of the actual manifests). This naming made the code extremely confusing, especially in code that deals with VMAgentManifest  and actual agent manifest. I renamed VMAgentManifest  to VMAgentFamily so you will see those renames throughout this PR.